tests/unit/test_application.py

```diff
@@ -154,15 +154,38 @@ def test_run_managed_specified(app, fake_project, emitter, monkeypatch):
         app.project.__class__,
         "get_build_plan",
         lambda _: [
-            BuildInfo(arch, "arch1", bases.BaseName("base", "1")),
-            BuildInfo(arch, "arch2", bases.BaseName("base", "2")),
+            BuildInfo("a1", arch, "arch1", bases.BaseName("base", "1")),
+            BuildInfo("a2", arch, "arch2", bases.BaseName("base", "2")),
         ],
     )
-    app.run_managed("arch2")
+    app.run_managed(None, "arch2")
 
-    emitter.assert_debug("Running testcraft in arch2 instance...")
+    emitter.assert_debug("Running testcraft:a2 in arch2 instance...")
     assert (
-        mock.call("debug", "Running testcraft in arch1 instance...")
+        mock.call("debug", "Running testcraft:a1 in arch1 instance...")
```

Comment:
this feels like an awkward way to check that only one of the build infos was used, particularly because if we change the message text this will keep passing (it's hard to assert negatives like this).
Shouldn't we instead check the calls to `mock_provider.instance()`? Then we can check that the `env` kwarg has the expected contents.

Reply:
Refactored to test actual calls instead of log output.